### PR TITLE
Fixes for clean_podcast_url()

### PR DESCRIPTION
**Pull request by mabdelfattah**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixes for the following URLs
```
https://pdst.fm/e/2.gum.fm/chtbl.com/track/chrt.fm/track/34D33/pscrb.fm/rss/p/traffic.megaphone.fm/ITLLC7765286967.mp3?updated=1687282661
https://pdst.fm/e/https://mgln.ai/e/441/www.buzzsprout.com/1121972/13019085-ep-252-the-deep-life-stack.mp3
```

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🧪🧹🎧

Improve `clean_podcast_url` function to handle more podcast URL formats and add corresponding tests. This change enhances the podcast extraction feature and ensures its reliability.

> _`Clean_podcast_url` is the master of the sound_
> _It strips away the noise and the lies that abound_
> _No matter what the platform or the service may try_
> _It finds the true essence of the podcast inside_

### Walkthrough
*  Update `clean_podcast_url` function to support more podcast URL formats ([link](https://github.com/yt-dlp/yt-dlp/pull/7556/files?diff=unified&w=0#diff-feda8f56946dc048e754111850baaef0eec4b9f8bbc2d3f04b1a785626ea5c0eL5124-R5130), [link](https://github.com/yt-dlp/yt-dlp/pull/7556/files?diff=unified&w=0#diff-feda8f56946dc048e754111850baaef0eec4b9f8bbc2d3f04b1a785626ea5c0eL5133-R5138))
* Add new test cases to `test_clean_podcast_url` function to cover the updated function ([link](https://github.com/yt-dlp/yt-dlp/pull/7556/files?diff=unified&w=0#diff-1780a2492dac55d0b4c5ff13cd3cf852a72cdac8995e17b10c87eeb678b3179eR1838-R1839))

</details>

### CODE REVIEW

File: yt_dlp/utils/_utils.py [hunk 1]
```diff
@@ -5121,16 +5121,21 @@ def clean_podcast_url(url):
     url = re.sub(r'''(?x)
         (?:
             (?:
-                chtbl\.com/track|
                 media\.blubrry\.com| # https://create.blubrry.com/resources/podcast-media-download-statistics/getting-started/
-                play\.podtrac\.com
+                play\.podtrac\.com|
+                chrt\.fm/track|
+                mgln.ai/e
```
Comment (bashonly) [inline yt_dlp/utils/_utils.py:5127]:
```suggestion
                mgln\.ai/e
```

File: yt_dlp/utils/_utils.py [hunk 2]
```diff
@@ -5121,16 +5121,21 @@ def clean_podcast_url(url):
...
             )/[^/]+|
+            (?:chtbl\.com/track)/[^\./]+|
+            (?:chtbl\.com/track)|
```
Comment (bashonly) [inline yt_dlp/utils/_utils.py:5130]:
```suggestion
            chtbl\.com/track(?:/[^/.]+)?|
```
&nbsp;
or, maybe it'd be better to move `chtbl\.com/track` back to where it was and do this:
```regex
            (?:
                chtbl\.com/track|
                media\.blubrry\.com| # https://create.blubrry.com/resources/podcast-media-download-statistics/getting-started/
                play\.podtrac\.com|
                chrt\.fm/track|
                mgln\.ai/e
            )(?:/[^/.]+)?|
```

File: yt_dlp/utils/_utils.py [hunk 3]
```diff
@@ -5121,16 +5121,21 @@ def clean_podcast_url(url):
...
             (?:dts|www)\.podtrac\.com/(?:pts/)?redirect\.[0-9a-z]{3,4}| # http://analytics.podtrac.com/how-to-measure
             flex\.acast\.com|
             pd(?:
                 cn\.co| # https://podcorn.com/analytics-prefix/
                 st\.fm # https://podsights.com/docs/
-            )/e
+            )/e|
+            [0-9]\.gum\.fm|
+            pscrb.fm/rss/p
```
Comment (bashonly) [inline yt_dlp/utils/_utils.py:5138]:
```suggestion
            pscrb\.fm/rss/p
```

**Review by bashonly — CHANGES_REQUESTED** [review]

**Comment by mabdelfattah** [comment]
Thanks @bashonly, for the code-review and feedback. I updated the code.

**Pull request merged**